appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/InvocationContext.java

```diff
@@ -64,8 +81,60 @@ public InvocationContext(ComponentInvocation invocation, ClassLoader contextClas
         this.contextClassLoader = contextClassLoader;
         this.securityContext = securityContext;
         this.useTransactionOfExecutionThread = useTransactionOfExecutionThread;
+        saveTracingContext();
     }
 
+    private void saveTracingContext() {
```

Comment:
Quite a high cyclomatic complexity on this method

Reply:
This is how it is on master - I'll raise a task to fix it separately.